### PR TITLE
fix: cleanup some more abe2e errors

### PR DESCRIPTION
**Pull request by cameronmeissner**
**What type of PR is this?**

/kind bug

**What this PR does / why we need it**:

cleans up some instance in abe2e where nil ptr dereferences were happening or had potential to occur

**Which issue(s) this PR fixes**:

Fixes #

**Requirements**:
- [ ] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests
- [ ] tested upgrade from previous version

**Special notes for your reviewer**:

**Release note**:
```
none
```

### CODE REVIEW

File: e2e/template.go [hunk 1]
```diff
@@ -576,3 +576,14 @@ type listVMSSVMNetworkInterfaceResult struct {
 		} `json:"properties,omitempty"`
 	} `json:"value,omitempty"`
 }
+
+func extractPrivateIP(res listVMSSVMNetworkInterfaceResult) (string, error) {
+	if res.Value != nil && len(res.Value) > 0 {
```
Comment (juan-lee) [inline e2e/template.go:581]: I believe a nil slice will return 0, so the nil check is redundant.

File: e2e/vmss.go [hunk 2]
```diff
@@ -110,7 +110,10 @@ func addPodIPConfigsForAzureCNI(vmss *armcompute.VirtualMachineScaleSet, vmssNam
 		}
 		podIPConfigs = append(podIPConfigs, ipConfig)
 	}
-	vmssNICConfig := vmss.Properties.VirtualMachineProfile.NetworkProfile.NetworkInterfaceConfigurations[0]
+	vmssNICConfig, err := getVMSSNICConfig(vmss)
+	if err != nil {
+		return fmt.Errorf("unable to get vmss nic: %s", err)
```
Comment (juan-lee) [inline e2e/vmss.go:115]: should you be wrapping the error with %w instead of %s?

File: e2e/vmss.go [hunk 3]
```diff
@@ -110,7 +110,10 @@ func addPodIPConfigsForAzureCNI(vmss *armcompute.VirtualMachineScaleSet, vmssNam
...
+	}
 	ipConfigs := append(vmssNICConfig.Properties.IPConfigurations, podIPConfigs...)
```
Comment (juan-lee) [inline e2e/vmss.go:117]: do you need ipConfigs? Can you just do append and assign on the same line?

File: e2e/vmss.go [hunk 4]
```diff
@@ -242,3 +249,26 @@ func getBaseVMSSModel(name, location, mcResourceGroupName, subnetID, sshPublicKe
...
+}
+
+func vmssModelHasVMNetworkProfile(vmss *armcompute.VirtualMachineScaleSet) bool {
+	return vmssModelHasVMProfile(vmss) && vmss.Properties.VirtualMachineProfile.NetworkProfile != nil
+}
+
+func getVMSSNICConfig(vmss *armcompute.VirtualMachineScaleSet) (*armcompute.VirtualMachineScaleSetNetworkConfiguration, error) {
+	if vmssModelHasVMNetworkProfile(vmss) {
+		networkProfile := vmss.Properties.VirtualMachineProfile.NetworkProfile
+		if networkProfile.NetworkInterfaceConfigurations != nil && len(networkProfile.NetworkInterfaceConfigurations) > 0 {
```
Comment (juan-lee) [inline e2e/vmss.go:268]: redundant nil check

File: e2e/vmss.go [hunk 5]
```diff
@@ -242,3 +249,26 @@ func getBaseVMSSModel(name, location, mcResourceGroupName, subnetID, sshPublicKe
...
+			nic := networkProfile.NetworkInterfaceConfigurations[0]
+			return nic, nil
```
Comment (juan-lee) [inline e2e/vmss.go:270]: `return networkProfile.NetworkInterfaceConfigurations[0], nil`

File: e2e/vmss.go [hunk 6]
```diff
@@ -242,3 +249,26 @@ func getBaseVMSSModel(name, location, mcResourceGroupName, subnetID, sshPublicKe
 		},
 	}
 }
+
+func vmssModelHasProperties(vmss *armcompute.VirtualMachineScaleSet) bool {
```
Comment (juan-lee) [inline e2e/vmss.go:253]: hasProperties, hasVMProfile, hasVMNetworkProfile? The parameter already gives a hint that it's a vmss model.

**Review by juan-lee — CHANGES_REQUESTED** [review]

File: e2e/template.go (hunk 1 above)
Comment (cameronmeissner) [reply e2e/template.go:581]: hah good to know

File: e2e/vmss.go (hunk 2 above)
Comment (cameronmeissner) [reply e2e/vmss.go:115]: we should actually go back and change this everywhere

File: e2e/vmss.go (hunk 3 above)
Comment (cameronmeissner) [reply e2e/vmss.go:117]: figured this was slightly more readable but can go ahead and change

File: e2e/vmss.go (hunk 6 above)
Comment (cameronmeissner) [reply e2e/vmss.go:253]: might just remove the helpers altogether

File: e2e/template.go (hunk 1 above)
Comment (juan-lee) [inline e2e/template.go:581]: redundant?

File: e2e/template.go [hunk 7]
```diff
@@ -576,3 +576,14 @@ type listVMSSVMNetworkInterfaceResult struct {
...
+		v := res.Value[0]
+		if v.Properties.IPConfigurations != nil && len(v.Properties.IPConfigurations) > 0 {
```
Comment (juan-lee) [inline e2e/template.go:583]: redundant?

**Review by juan-lee — CHANGES_REQUESTED** [review]

**Review by juan-lee — APPROVED** [review]

**Review by SeanDougherty — APPROVED** [review]

**Pull request merged**